### ISSUE TITLE
Handle multiple spectral windows

### ISSUE THREAD

**Issue by sjperkins** [ska-sa/montblanc#78]
At present, Montblanc only loads in the first spectral window from a MeasurementSet.

# Basic Specification

- The number of channels in each spectral window should be summed to produce a total.
- The frequencies from each spectral window should be loaded into their appropriate channel.
- Only **one** reference frequency/wavelength will be supported.

# Changes required

Probably need to make loader changes
- in the [header](https://github.com/ska-sa/montblanc/blob/036009805348419f259568298ec8433acc6a133d/montblanc/impl/common/loaders/loaders.py#L52) loader.
- in the [loader](https://github.com/ska-sa/montblanc/blob/036009805348419f259568298ec8433acc6a133d/montblanc/impl/biro/v2/loaders/loaders.py#L52-L59).

@saiyanprince Take a look at the above. Will you simply take the first reference frequency in the MS as the default?

EDIT One: Added a Basic Specification

**Comment by marziarivi** [comment]
Actually, I need this!

**Comment by sjperkins** [comment]
@marziarivi I think @saiyanprince will be making some changes here. Feel free to make suggestions and we can update the above specification.

**Comment by marziarivi** [comment]
Actually I am not used with MS, but for my simulations it would be helpful to specify a frequency band split in several channel of a given bandwidth. At the moment I use MS only for reading the uvw coordinates but then I would like to give as an independent parameter frequency specifications, so that this information is not related to the MS I am using and I can change it.

**Comment by jtlz2** [comment]
@marziarivi Soon we can probably get someone (A. Student) to add a function
to generate uvw coordinates from observing parameters rather than take them
from an MS - will keep you updated.

**Comment by marziarivi** [comment]
Yes, but you have to read the antennas position anyway.

**Comment by jtlz2** [comment]
@marziarivi - of course - but that's << MS in size.

**Comment by marziarivi** [comment]
Hi,
looking at the sections of code mentioned above in order to deal with several channels, I don't see any change required. It seems it already manages more channels read from the MS. Am I wrong?
It would be also useful to have the possibility to set the number of channels and their frequency manually.

**Comment by sjperkins** [comment]
Currently it only reads channels from the first spectral window. Note the `[0]` index below:

```python
# Determine the wavelengths
freqs = tf.getcol('CHAN_FREQ')
wavelength = (montblanc.constants.C/freqs[0]).astype(solver.ft)
solver.transfer_wavelength(wavelength)

# First dimension also seems to be of size 1 here...
# Divide speed of light by frequency to get the wavelength here.
solver.set_ref_wave(montblanc.constants.C/tf.getcol('REF_FREQUENCY')[0])
```
but the loader calculates the number of channels from the number of elements in the `CHAN_FREQ` column.

```python
 nchan = self.tables['freq'].getcol('CHAN_FREQ').size
```

I think this will result in counting all channels from all spectral windows, so there's probably a subtle bug here.

I can easily enable the ability to create an "empty" solver, where you manually specify the number of timesteps, antenna, channels etc. But the user will then be responsible for specifying all the configuration thats not in the MS.

**Comment by marziarivi** [comment]
I think there is no need if I can just change the nchan variable and set the central wavelengths of each channel after reading the MS. But how can I set the channel bandwidth?

**Comment by sjperkins** [comment]
There is no ability to change the number of channels after the solver is created. So that will need to specified when configuring the solver. Let me know if you want to go this route, it's a quick change.

However, you can change the frequencies and reference frequency after solver creation. If you had `nchan=64` channels:

```python
slvr.transfer_frequencies(np.linspace(1e9, 2e9, 64))
slvr.set_ref_freq(1.5e9)
```

**Comment by sjperkins** [comment]
Implemented in #140
